tests/k8st/tests/test_bgp_advert.py

```diff
@@ -129,8 +129,24 @@ def tearDown(self):
     def get_svc_cluster_ip(self, svc, ns):
         return run("kubectl get svc %s -n %s -o json | jq -r .spec.clusterIP" % (svc, ns)).strip()
 
-    def do_external_curl(self, container, hostname):
-        return run("docker exec %s curl --connect-timeout 2 -m 3 %s" % (container, hostname))
+    def do_external_curl(self, container, hostname, succeed=True):
+        out = "failed"
+        try:
+          for i in range(10):
```

Comment:
I'm.. not a fan of adding the loop into `do_external_curl`.

I think there are a lot of situations where we don't want this, and it's better to allow the calling code to make the decision as to whether or not it wants to retry at all, and if so how many times.

Can we revert this change and just loop over the call within the test code itself?